common/guestchain/src/manager.rs

```diff
@@ -233,6 +219,36 @@ impl<PK: crate::PubKey> ChainManager<PK> {
         Ok(epoch_ends)
     }
 
+    pub fn validate_generate_next(
+        &self,
+        host_height: crate::HostHeight,
+        host_timestamp: NonZeroU64,
+        state_root: &CryptoHash,
+        force: bool,
+    ) -> Result<(), GenerateError> {
+        if self.pending_block.is_some() {
+            return Err(GenerateError::HasPendingBlock);
+        }
+        if !host_height.check_delta_from(
+            self.header.host_height,
+            self.config.min_block_length,
+        ) {
+            return Err(GenerateError::BlockTooYoung);
+        }
+
+        let next_epoch = self.maybe_generate_next_epoch(host_height);
```

Comment:
I don’t like this approach because it generates the next epoch twice. Perhaps change validate_generate_next to return the epoch?

Reply:
yes that could work